src/storage/src/source/persist_source.rs

```diff
@@ -132,91 +128,42 @@ where
             let mut current_ts = 0;
 
             move |cap, output| {
-                let waker = futures_util::task::waker_ref(&activator);
                 let mut context = Context::from_waker(&waker);
-
-                // Bound execution of operator to prevent a single operator from hogging
-                // the CPU if there are many messages to process
-                let timer = Instant::now();
-
                 while let Poll::Ready(item) = pinned_stream.as_mut().poll_next(&mut context) {
                     match item {
-                        Some(Ok(event)) => match event {
-                            ListenEvent::Progress(upper) => {
-                                // NOTE(aljoscha): Ideally, we would only get a CapabilitySet, that
-                                // we have to manage. Or the ability to drop the plain Capability
-                                // right at the start.
-                                if let Some(first_element) = upper.first() {
-                                    current_ts = *first_element;
-                                    cap.downgrade(first_element);
-                                }
+                        Some(Ok(ListenEvent::Progress(upper))) => {
+                            if upper.elements().is_empty() {
+                                return SourceStatus::Done;
                             }
-                            ListenEvent::Updates(updates) => {
-                                let cap = cap.delayed(&current_ts);
-                                let mut session = output.session(&cap);
-                                for update in updates {
-                                    session.give(Ok(update));
-                                }
+                            if let Some(first_element) = upper.first() {
+                                current_ts = *first_element;
+                                cap.downgrade(first_element);
```

Comment:
should we have a comment here (and in `build_storage_dataflow`) on when we assume 1d time?

Reply:
Good point, I changed this to use `antichain.into_option()` which only exists for timestamps implementing `timely::order::TotalOrder` (i.e they are 1d). This will turn into a compile time error if we ever make this generic